build_tools/cmake/build_docs.sh

```diff
@@ -0,0 +1,54 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Build the IREE documentation with CMake. Designed for CI, but can be run
+# manually. This uses previously cached build results and does not clear the
+# build directory.
+
+set -x
+set -e
+
+ROOT_DIR=$(git rev-parse --show-toplevel)
+BUILD_DIR=${BUILD_DIR:-"build-docs"}
+
+CMAKE_BIN=${CMAKE_BIN:-$(which cmake)}
+
+"$CMAKE_BIN" --version
+ninja --version
+
+cd ${ROOT_DIR?}
+if [ -d ${BUILD_DIR} ]
+then
+  echo "Build directory already exists. Will use cached results there."
+else
+  echo "Build directory does not already exist. Creating a new one."
+  mkdir ${BUILD_DIR}
+fi
+cd ${BUILD_DIR}
+
+"$CMAKE_BIN" .. -DCMAKE_BUILD_TYPE=Release \
+                -DIREE_BUILD_COMPILER=ON \
+                -DIREE_BUILD_TESTS=ON \
```

Comment:
Why build the compiler and tests? Is `IREE_BUILD_DOCS=ON` sufficient?

Reply:
No. If turning of `IREE_BUILD_COMPIER` then `iree/compiler/` directory won't be included. That's where the dialects reside. For `IREE_BUILD_TESTS` I think there is a bug at the moment at https://github.com/google/iree/blob/master/build_tools/cmake/iree_cc_embed_data.cmake#L88 and https://github.com/google/iree/blob/master/build_tools/cmake/iree_spirv_kernel_cc_library.cmake#L61. `TESTONLY` should be an option taking no parameters? So effectively we are marking all `iree_cc_embed_data` and `iree_spirv_kernel_cc_library` as test only. But I might misunderstand something. + @marbre to confirm.

Reply:
Filed #1330 to track this.

Reply:
Yeah, you need to set `IREE_BUILD_COMPIER` such that `add_directory` is called. And yes, after a real quick look, it seems like a bug regarding the pass through of `TESTONLY`. We probably always build with tests enabled at the moment and thus didn't noticed. A follow up issue to track would be great!